src/zmq/zmqpublishnotifier.cpp

```diff
@@ -234,6 +246,27 @@ bool CZMQPublishRawBlockNotifier::NotifyBlock(const CBlockIndex *pindex)
     return SendMessage(MSG_RAWBLOCK, &(*ss.begin()), ss.size());
 }
 
+bool CZMQPublishRawChainLockNotifier::NotifyChainLock(const CBlockIndex *pindex)
+{
+    LogPrint("zmq", "zmq: Publish rawchainlock %s\n", pindex->GetBlockHash().GetHex());
+
+    const Consensus::Params& consensusParams = Params().GetConsensus();
+    CDataStream ss(SER_NETWORK, PROTOCOL_VERSION);
+    {
+        LOCK(cs_main);
+        CBlock block;
+        if(!ReadBlockFromDisk(block, pindex, consensusParams))
+        {
+            zmqError("Can't read block from disk");
+            return false;
+        }
+
+        ss << block;
```

Comment:
Doesn't this push the whole block instead of just the header?

Reply:
Yes, why?

Reply:
Oh, I see 🙈 fixed the description 😊

Reply:
Isn't this overkill? I'd assume that listeners already listen for blocks through other notifications, so this would usually generate duplicate data?

Reply:
The use cases for these "lock" notifications is when you want to ignore "normal" notifications in some way and at the same time you want to avoid querying rpc to get the actual data. For example, if you need smth like "I'm only interested in locked blocks" - use `rawchainlock` (you don't need `rawblock` in this case). If instead you want smth like "I'm interested in all blocks and also I want to know when some of them are locked" then you need a `rawblock`+`hashchainlock` combination (should be trivial to calculate the hash from the block itself and match them).

Reply:
Let's ping @snogcel to see if I understand these use cases correctly :)

Reply:
@UdjinM6 summarized the use cases perfectly - rawchainlock provides a lot of data but can be thought of as a replacement for the hashchainlock + getblock (JSON-RPC) combination you'd see otherwise.

Reply:
ok, understood